example.js

```diff
@@ -28,21 +38,56 @@ import {
 import otpConfig from './config.yml'
 
 // Define icon sets for modes.
-const MyLegIcon = TriMetLegIcon
-const MyModeIcon = TriMetModeIcon
+const MyLegIcon = ClassicLegIcon
+const MyModeIcon = ClassicModeIcon
+
 /**
  * For testing, try uncommenting the following two statements (and comment the two above),
  * and see how the icons get changed in:
  * - the mode options panel (select transit, bike+transit, etc.)
  * - the itinerary narrative (step-by-step directions).
- *
- * For more advanced testing, you can replicate and customize components and observe the change in icons.
- * - For LegIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-leg-icon.js
- * - For ModeIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-mode-icon.js
  */
 // const MyLegIcon = () => <Ferry />
 // const MyModeIcon = () => <AerialTram />
 
+/** 
+ * For more advanced users, you can replicate and customize components and observe the change in icons.
+ * - For LegIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-leg-icon.js
+ * - For ModeIcon: https://github.com/opentripplanner/otp-ui/blob/master/packages/icons/src/trimet-mode-icon.js
+ * The example below shuffles some icons around for demonstration purposes.
+ */
+/*
+const CustomTransitIcon = Ferry
+const CustomRailIcon = ClassicGondola
+const CustomStreetcarIcon = StandardGondola
+const CustomBikeRentalIcon = ClassicBus
+
+const MyModeIcon = ({ mode, ...props }) => {
+  if (!mode) return null;
+  switch (mode.toLowerCase()) {
+    // Place custom icons for each mode here.
+    case "transit":
+      return <CustomTransitIcon {...props} />
+    case "rail":
+      return <CustomRailIcon {...props} />
+    default:
+      return <ClassicModeIcon mode={mode} {...props} />
+  }
+}
+
+const MyLegIcon = ({ leg, ...props }) => {
+  if (
+    leg.routeLongName &&
+    leg.routeLongName.startsWith('MAX')
+  ) {
+    return <CustomStreetcarIcon />
+  } else if (leg.rentedBike) {
+    return <CustomBikeRentalIcon />
+  }
+  return <LegIcon leg={leg} ModeIcon={MyModeIcon} {...props} />
+}
+*/
+
```

Comment:
@binh-dam-ibigroup, this is growing to be a lot to include in the `example.js` file. I think it might make more sense to move this to a `custom-icons.js` file and enable playing around with icon overrides with a single change. I've provided some changes in pursuit of this in https://github.com/opentripplanner/otp-react-redux/pull/164.

Reply:
Good idea.